frontend/packages/kubevirt-plugin/src/components/vms/vm-activity.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardLink from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardLink';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import ActivityBody, {
+  RecentEventsBody,
+} from '@console/shared/src/components/dashboard/activity-card/ActivityBody';
+import { getName, getNamespace } from '@console/shared';
+import { resourcePath, FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventModel } from '@console/internal/models';
+import { EventKind } from '@console/internal/module/k8s';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { VMKind } from '../../types';
+import { VirtualMachineModel } from '../../models';
+import { getVmEventsFilters } from '../../selectors/event';
+import { VMDashboardContext } from './vm-dashboard-context';
+
+const combinedVmFilter = (vm: VMKind) => {
```

Comment:
it would be more descriptive to add return type directly here and then you wouldnt have to add types to the function you are returning

Reply:
done, return types are added